rust/cardano-blockchain-types/src/auxdata/aux_data.rs

```diff
@@ -0,0 +1,170 @@
+                Ok(TransactionAuxData {
+                    metadata: Metadata::decode(d, &mut ())?,
+                    scripts: TransactionScripts::default(),
+                })
+            },
+            // Shelley-MA: https://github.com/IntersectMBO/cardano-ledger/blob/78b32d585fd4a0340fb2b184959fb0d46f32c8d2/eras/conway/impl/cddl-files/conway.cddl#L523
+            Ok(minicbor::data::Type::Array) => Self::decode_shelley_ma_array(d),
+            // Maybe Alonzo and beyond: https://github.com/IntersectMBO/cardano-ledger/blob/78b32d585fd4a0340fb2b184959fb0d46f32c8d2/eras/conway/impl/cddl-files/conway.cddl#L526
+            Ok(minicbor::data::Type::Tag) => Self::decode_alonzo_plus_map(d),
+            Ok(unexpected) => {
+                let msg = format!(
+                    "Error decoding Transaction Aux Data: Unexpected datatype {unexpected}"
+                );
+                Err(minicbor::decode::Error::message(&msg))
+            },
+            Err(error) => {
+                let msg = format!("Error decoding Transaction Aux Data: {error}");
+                Err(minicbor::decode::Error::message(msg))
+            },
+        }
+    }
+}
+
+impl TransactionAuxData {
+    /// Get metadata with the given label.
+    #[must_use]
+    pub fn metadata(&self, label: MetadatumLabel) -> Option<&MetadatumValue> {
+        self.metadata.get(label)
+    }
+
+    /// Decode a Shelley-MA Auxiliary Data Array
+    fn decode_shelley_ma_array(d: &mut minicbor::Decoder) -> Result<Self, minicbor::decode::Error> {
+        match d.array() {
+            Ok(Some(entries)) => {
+                if entries != 2 {
+                    let msg = format!(
+                        "Error decoding Transaction Aux Data: Script Data Array Expected 2 entries, found {entries}."
+                    );
+                    return Err(minicbor::decode::Error::message(&msg));
+                }
+            },
+            Ok(None) => {
+                return Err(minicbor::decode::Error::message(
+                    "Error decoding Transaction Aux Data: Indefinite Array found decoding Metadata. Invalid."));
+            },
+            Err(error) => {
+                return Err(minicbor::decode::Error::message(format!(
+                    "Error decoding Transaction Aux Data: {error}."
+                )));
+            },
+        };
+
+        let metadata = Metadata::decode(d, &mut ())?;
+        let script_array = ScriptArray::decode(d, &mut ScriptType::Native)?;
+
+        let scripts = MutableTransactionScriptsMap::default();
+        scripts.insert(ScriptType::Native, script_array);
+
+        Ok(Self {
+            metadata,
+            scripts: scripts.into(),
+        })
+    }
+
+    /// Decode an Alonzo Plus MAP
+    fn decode_alonzo_plus_map(d: &mut minicbor::Decoder) -> Result<Self, minicbor::decode::Error> {
+        match d.tag() {
+            Ok(tag) => {
+                if tag.as_u64() != 259 {
```

Comment:
What is this 259?

Reply:
Comes from here: https://github.com/IntersectMBO/cardano-ledger/blob/78b32d585fd4a0340fb2b184959fb0d46f32c8d2/eras/conway/impl/cddl-files/conway.cddl#L526
Could be worth adding a comment.
It's not a registered or standard CBOR Tag.